frontend/src/scenes/dashboard/dashboardLogic.tsx

```diff
@@ -1077,18 +1083,26 @@ export const dashboardLogic = kea<dashboardLogicType>({
                 actions.refreshAllDashboardItems({ action: 'refresh_above_threshold', initialLoad, dashboardQueryId })
                 allLoaded = false
             } else {
-                const notYetLoadedItems = values.tiles?.filter(
-                    (t): t is DashboardTile => !!t.insight && !t.insight.result
-                )
-                if (notYetLoadedItems && notYetLoadedItems?.length > 0) {
+                const tilesWithNoResults = values.tiles?.filter((t) => !!t.insight && !t.insight.result) || []
+                const tilesWithResults = values.tiles?.filter((t) => !!t.insight && t.insight.result) || []
+
+                if (tilesWithNoResults && tilesWithNoResults?.length > 0) {
                     actions.refreshAllDashboardItems({
-                        tiles: notYetLoadedItems,
+                        tiles: tilesWithNoResults,
                         action: 'load_missing',
                         initialLoad,
                         dashboardQueryId,
                     })
                     allLoaded = false
                 }
+
+                tilesWithResults
+                    ?.map((t) => t.insight)
+                    .filter((i): i is InsightModel => !!i)
+                    .forEach((i) => {
```

Comment:
Q:Why are we opting for `forEach` instead of a for loop here? The code looked initially more complicated than it should